cli/triggers.md

```diff
@@ -0,0 +1,112 @@
+# Lamba Triggers
+
+Certain AWS Services can [invoke Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-services.html) in response to lifecycle events.  The Amplify CLI provides trigger templates for common use cases.
+
+If you wish to modify the functionality of these templates, you are able to do so locally before pushing them.
+
+## Auth
+
+The CLI Auth workflow provides the following Lambda trigger templates:
+
+### Custom Auth Challenge with Google reCaptcha
+
+Captchas allow front end applications to guards against bots or other unwanted page interactions by presenting a challenge that is designed to require human intervention.  The Google reCaptcha service is a popular implementation of captcha.  
+
+This template will configure three triggers: CreateAuthChallenge, DefineAuthChallenge and VerifyAuthChallengeResponse.
+
+The first two will essentially allow the standard username/password flow to execute unimpeded, while VerifyAuthChallengeResponse will accept data from your client-side captcha component and verify it against Google's reCaptcha service via axios.
```

Comment:
We should a link to axios.

Reply:
We shouldn't actually mention Axios, instead we should describe what is happening - e.g. when `Auth.sendCustomChallengeAnswer ` runs it invokes the VerifyAuthChallengeResponse in Cognito which calls a Lambda function that executes an HTTP request to Google, thus validating the captcha on the backend and returning the result to the client.

Reply:
Done.